core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -866,6 +930,7 @@ final class ShuffleBlockFetcherIterator(
                 }
 
                 if (blockId.isShuffleChunk) {
+                  shuffleMetrics.incCorruptMergedBlockChunks(1)
```

Comment:
We need to add this for the invocation of `initiateFallbackFetchForPushMergedBlock` [above](https://github.com/apache/spark/blob/72cce5c39da8a52efa0a7cad4d65dac4eb389ff6/core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala#L806) as well - the zero byte read is unexpected, and is essentially a corrupt read

Reply:
Thanks for calling this out, added this change.